comp/api/api/apiimpl/internal/agent/agent.go

```diff
@@ -99,7 +101,7 @@ func SetupHandlers(
 	r.HandleFunc("/{component}/status", componentStatusGetterHandler).Methods("GET")
 	r.HandleFunc("/{component}/status", componentStatusHandler).Methods("POST")
 	r.HandleFunc("/{component}/configs", componentConfigHandler).Methods("GET")
-	r.HandleFunc("/gui/csrf-token", getCSRFToken).Methods("GET")
+	r.HandleFunc("/gui/csrf-token", func(w http.ResponseWriter, _ *http.Request) { getCSRFToken(w, gui) }).Methods("GET")
```

Comment:
Once @jeremy-hanna work on being able to register routes into the API server using Fx value groups we would be able to get rid of having to declare component like GUI as a dependency 

cc @ogaca-dd

Reply:
Indeed, that is the goal of using fx value group for the API Server.

Reply:
I am wondering if not having `"/gui/csrf-token"` behaves the same as having an endpoint `"/gui/csrf-token"` that returns an error.